responders/src/UI/embc-responder/src/app/feature-components/wizard/wizard-adapter.service.ts

```diff
@@ -71,37 +75,58 @@ export class WizardAdapterService {
       dateOfBirth: this.evacueeSearchService.evacueeSearchContext
         ?.evacueeSearchParameters?.dateOfBirth
     };
+
+    this.stepEvacueeProfileService.profileTabs = this.wizardDataService.createNewProfileSteps();
   }
 
   /**
    * Set initial values for Edit Registrant Profile (stepProfileService), from an Evacuee Profile ID
    */
-  public stepEditProfileFromProfileId(profileId: string) {
-    this.evacueeProfileService
-      .getProfileFromId(profileId)
-      .subscribe((registrantProfileModel) => {
-        this.stepEvacueeProfileService.setFormValuesFromProfile(
-          registrantProfileModel
-        );
-        this.stepEvacueeProfileService.setEditTabStatus();
-      });
+  public stepEditProfileFromProfileId(profileId: string): Observable<boolean> {
+    return new Observable<boolean>((obs) => {
+      this.evacueeProfileService.getProfileFromId(profileId).subscribe(
+        (registrantProfileModel: RegistrantProfileModel) => {
+          this.stepEvacueeProfileService.setFormValuesFromProfile(
+            registrantProfileModel
+          );
+
+          this.stepEvacueeProfileService.profileTabs = this.wizardDataService.createNewProfileSteps();
+          this.stepEvacueeProfileService.setEditTabStatus();
+
+          obs.next(true);
+        },
+        (error) => {
```

Comment:
We need to think about the path of failure as well for all the api integrations in adapter service. If the api fails to load data, we can still load the wizard and show the error message there. Thoughts?

Reply:
Usually, the page needs whatever data we're requesting, but we should figure out a way to display to the user that an error occurred, instead of silently refusing the navigation. A general healthcheck was implemented on the Supplier portal, which would account for most situations where this request would fail.